### PR TITLE
Fix tests in XCode 10

### PR DESCRIPTION
**Pull request by janhalousek**
Fixes failing tests in XCode 10

This MR is superset of https://github.com/Swinject/Swinject/pull/369 and probably resolves issues #368, #367 (haven't tested all cases).

The problem lies in casting nil variable to _Any_ type, which seems to be a new thing in XCode 10 / Swift 4.2
```
class Bar {}
var foo: Bar? = nil

if let fooUnwrapped = foo as? Any {
    // this passes and fooUnwrapped is Optional(nil)
}
```

### CODE REVIEW

File: Sources/InstanceStorage.swift [hunk 1]
```diff
@@ -120,7 +120,10 @@ private class Weak<Wrapped> {
     }
 #else
     var value: Wrapped? {
-        get { return object as? Wrapped }
+        get {
+            guard let object = object else { return nil }
+            return object as? Wrapped
+        }
```
Comment (jakubvano) [inline Sources/InstanceStorage.swift]: This fix also needs to be applied to the linux branch

Comment (janhalousek) [reply Sources/InstanceStorage.swift]: Adjusted in ce3b62e702d8fb59a41999353c8a257878adc035

Comment (jakubvano) [reply Sources/InstanceStorage.swift]: 👍

**Comment by jakubvano** [comment]
Thanks for the PR! 👍 
I will try to release Swift 4.2 compatible version ASAP

**Pull request merged**